uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/leave-calendar-admin.controller.js

```diff
@@ -60,6 +70,46 @@ define([
     }
 
     /**
+     * Load all contracts or retrieve them from cache
+     *
+     * @return {Promise}
+     */
+    function loadContracts () {
+      if (contracts) {
+        return $q.resolve(contracts);
+      }
+
+      return Contract.all();
+    }
+
+    /**
+     * Get contact IDs filtered according to contracts that belong
+     * to the currently selected absence period
+     *
+     * @return {Promise}
+     */
+    function getContactIdsToReduceTo () {
+      return loadContracts()
+      .then(function (contracts) {
+        var contractsInAbsencePeriod = contracts.filter(function (contract) {
+          var details = contract.info.details;
+          var isContractInAbsencePeriod = (
```

Comment:
instead of:
```js
var isContractInAbsencePeriod = (
  (!details.period_start_date ||
  moment(details.period_start_date).diff(moment(vm.selectedPeriod.end_date)) <= 0) &&
  (!details.period_end_date ||
  moment(details.period_end_date).diff(moment(vm.selectedPeriod.start_date)) > 0)
);
```

you could try:
```js
var isContractInAbsencePeriod = (
  moment(details.period_start_date).isSameOrBefore(vm.selectedPeriod.end_date) &&
  moment(details.period_end_date).isAfter(vm.selectedPeriod.start_date)
);
```
(And why > 0 instead of >= 0?)

Reply:
Unfortunately, Contract may **not have** `end_date`.

![image](https://user-images.githubusercontent.com/3973243/29715744-813e2a44-89a0-11e7-8e47-ab96632d1ee7.png)

And moment will return `false`, while I it needs to return `true` since if there is no defined end date - the contract automatically falls into the selected absence period.

![image](https://user-images.githubusercontent.com/3973243/29715707-52d2c85e-89a0-11e7-9849-ac373549339c.png)